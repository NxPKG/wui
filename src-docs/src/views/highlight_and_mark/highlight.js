import React, { Fragment, useState } from 'react';

import {
  WuiHighlight,
  WuiFieldSearch,
  WuiFormRow,
  WuiSpacer,
  WuiSwitch,
} from '../../../../src/components';

export function Highlight() {
  const [searchValue, setSearchValue] = useState('jumped over');
  const [isHighlightAll, setHighlightAll] = useState(false);

  const onSearchChange = e => {
    setSearchValue(e.target.value);
  };
  const changeHighlightAll = e => {
    setHighlightAll(e.target.checked);
  };

  return (
    <Fragment>
      <WuiFormRow label="Enter text to highlight substrings within a string">
        <WuiFieldSearch
          value={searchValue}
          onChange={e => {
            onSearchChange(e);
          }}
        />
      </WuiFormRow>

      <WuiSpacer size="m" />
      <WuiSwitch
        label="Highlight all"
        checked={isHighlightAll}
        onChange={e => changeHighlightAll(e)}
      />
      <WuiSpacer size="m" />
      <WuiHighlight search={searchValue} highlightAll={isHighlightAll}>
        The quick brown fox jumped over the lazy dog
      </WuiHighlight>
    </Fragment>
  );
}
