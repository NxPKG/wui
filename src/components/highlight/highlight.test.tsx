/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { WuiHighlight } from './highlight';

describe('WuiHighlight', () => {
  test('is rendered', () => {
    const component = render(
      <WuiHighlight {...requiredProps} search="">
        value
      </WuiHighlight>
    );

    expect(component).toMatchSnapshot();
  });

  describe('behavior', () => {
    describe('matching', () => {
      test('only applies to first match', () => {
        const component = render(
          <WuiHighlight search="match">match match match</WuiHighlight>
        );

        expect(component).toMatchSnapshot();
      });

      test('applies to all matches', () => {
        const component = render(
          <WuiHighlight search="match" highlightAll>
            match match match
          </WuiHighlight>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('loose matching', () => {
      test('matches strings with different casing', () => {
        const component = render(
          <WuiHighlight search="CASE">different case match</WuiHighlight>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('strict matching', () => {
      test("doesn't match strings with different casing", () => {
        const component = render(
          <WuiHighlight search="CASE" strict>
            different case match
          </WuiHighlight>
        );

        expect(component).toMatchSnapshot();
      });
    });
  });
});
