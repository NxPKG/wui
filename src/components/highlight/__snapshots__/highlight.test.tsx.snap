// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WuiHighlight behavior loose matching matches strings with different casing 1`] = `
<span>
  different 
  <mark
    class="wuiMark"
  >
    case
  </mark>
   match
</span>
`;

exports[`WuiHighlight behavior matching applies to all matches 1`] = `
<span>
  <mark
    class="wuiMark"
  >
    match
  </mark>
   
  <mark
    class="wuiMark"
  >
    match
  </mark>
   
  <mark
    class="wuiMark"
  >
    match
  </mark>
</span>
`;

exports[`WuiHighlight behavior matching only applies to first match 1`] = `
<span>
  <mark
    class="wuiMark"
  >
    match
  </mark>
   match match
</span>
`;

exports[`WuiHighlight behavior strict matching doesn't match strings with different casing 1`] = `
<span>
  different case match
</span>
`;

exports[`WuiHighlight is rendered 1`] = `
<span
  aria-label="aria-label"
  class="testClass1 testClass2"
  data-test-subj="test subject string"
>
  value
</span>
`;
